File: Data Structures & Algorithms/number-of-one-bits/submission-0.cpp
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int count = 0;
        int bit;

        while (n > 0)
        {
            bit = n & 1;
            if (bit == 1)
            {
                count++;
            }

            n = n >> 1;
        }

        return count;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def getCharCount(self, s: str):
        count = [0] * 26
        
        for c in s:
            count[ord(c) - ord("a")] += 1
        
        return tuple(count)

    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagramDict = defaultdict(list)

        for s in strs:
            anagramDict[self.getCharCount(s)].append(s)

        return anagramDict.values()

